### PR TITLE
issue-25 Respect IHandlerResolver's responsibility to create handlers for every incoming message.

### PR DESCRIPTION
**Pull request by payman81**

### CODE REVIEW

File: JustSaying/JustSayingFluently.cs [hunk 1]
```diff
@@ -231,16 +231,18 @@ public IFluentSubscription IntoQueue(string queuename)
...
+            if(!handlerResolver.ResolveHandlers<T>().Any())
             {
                 throw new HandlerNotRegisteredWithContainerException(string.Format("IHandler<{0}> is not regsistered in the container.", typeof(T).Name));
             }
-            foreach (var handler in handlers)
+            if (handlerResolver.ResolveHandlers<T>().Count() > 1)
             {
-                Bus.AddMessageHandler(handler);
-                Log.Info(string.Format("Added a message handler - Topic: {0}, QueueName: {1}, HandlerName: {2}", _subscriptionConfig.Topic, _subscriptionConfig.QueueName, handler.GetType().Name));
+                throw new NotSupportedException(string.Format("There are more than one registration for IHandler<{0}>. JustSaying currently does not support multiple registration for IHandler<T>.", typeof(T).Name));
```
Comment (payman81) [inline JustSaying/JustSayingFluently.cs]: Allowing multiple registration requires a lot of thinking and perhaps major redesign. This can be done later.

Comment (antonjefcoate) [inline JustSaying/JustSayingFluently.cs]: Just to clarify, does this mean multiple handlers of the same message can no longer be applied?
OR, is it just when using IOC style handler resolution that the limitation is imposed?

**Comment by antonjefcoate** [comment]
Aside from that, all looks good...

File: JustSaying/JustSayingFluently.cs (hunk 1 above)
Comment (payman81) [inline JustSaying/JustSayingFluently.cs]: Yes that restriction is when you use IoC only.

**Pull request merged**